server/server.go

```diff
@@ -65,27 +42,28 @@ func New(config Config) (microserver.Server, error) {
 		}
 	}
 
-	newServer := &server{
-		// Dependencies.
-		logger: config.MicroServerConfig.Logger,
-
-		// Internals.
-		bootOnce:     sync.Once{},
-		config:       config.MicroServerConfig,
+	newServer := &Server{
+		logger: config.Logger,
+
+		bootOnce: sync.Once{},
+		config: microserver.Config{
+			Logger:      config.Logger,
+			ServiceName: config.ProjectName,
+			Viper:       config.Viper,
+
+			Endpoints: []microserver.Endpoint{
+				endpointCollection.Healthz,
+				endpointCollection.Version,
+			},
+			ErrorEncoder: errorEncoder,
+		},
 		shutdownOnce: sync.Once{},
 	}
 
-	// Apply internals to the micro server config.
-	newServer.config.Endpoints = []microserver.Endpoint{
-		endpointCollection.Healthz,
-		endpointCollection.Version,
-	}
-	newServer.config.ErrorEncoder = newServer.newErrorEncoder()
-
 	return newServer, nil
 }
 
-type server struct {
+type Server struct {
```

Comment:
I don't think Server needs to be exported. But I wouldn't mind making it explicit return type of `New`.

Reply:
Server is no longer exported. I left the return type as `microserver.Server` because its the interface.